src/NServiceBus.AcceptanceTests/EndpointTemplates/ConfigureExtensions.cs

```diff
@@ -63,38 +37,43 @@ public static void DefineTransactions(this BusConfiguration config, IDictionary<
             }
         }
 
-        public static async Task DefinePersistence(this BusConfiguration config, IDictionary<string, string> settings)
+        public static Task DefinePersistence(this BusConfiguration config, IDictionary<string, string> settings)
         {
             if (!settings.ContainsKey("Persistence"))
             { 
                 settings = Persistence.Default.Settings;
             }
 
-            var persistenceType = Type.GetType(settings["Persistence"]);
+            return ConfigureTestExecution(TestDependencyType.Persistence, config, settings);
+        }
+
+        enum TestDependencyType
+        {
+            Transport,
+            Persistence
+        }
+
+        private static async Task ConfigureTestExecution(TestDependencyType type, BusConfiguration config, IDictionary<string,string> settings)
+        {
+            var dependencyTypeString = type.ToString();
 
+            var dependencyType = Type.GetType(settings[dependencyTypeString]);
 
-            var typeName = "Configure" + persistenceType.Name;
+            var typeName = "Configure" + dependencyType.Name;
```

Comment:
With my above "List<>" suggestion we could make this just look for loaded types assignable to `IConfigureTestExecution` ?